app/assets/javascripts/user.js

```diff
@@ -178,6 +220,48 @@ $(document).ready(function () {
     }
   }
 
+  function updateHomeLocation() {
+    const lat = $("#home_lat").val().trim();
+    const lon = $("#home_lon").val().trim();
+    if (!lat || !lon) {
+      return;
+    }
+
+    const geocodeUrl = "/geocoder/search_osm_nominatim_reverse";
+    const params = {
+      format: "json",
+      lat,
+      lon,
+      zoom: 3
+    };
+
+    if (locationInput.request) locationInput.request.abort();
+    locationInput.request = $.ajax({
+      url: geocodeUrl,
+      method: "POST",
+      data: params,
+      success: function (country) {
+        locationInput.request = null;
+        locationInput.countryName = country;
+        $("#location_default_name").text(I18n.t("javascripts.profiles.edit.location_autofill", { country }));
+        if (locationInput.dirty) {
+          updateLocationWarning();
+        } else {
+          $("#location_name").val(locationInput.countryName);
+        }
+      }
+    });
+  }
+
+  function updateLocationWarning() {
+    const locationName = $("#location_name").val();
+    if (!locationInput.dirty || locationName.includes(locationInput.countryName)) {
+      $("#location_name_warning").addClass("d-none");
+    } else {
+      $("#location_name_warning").removeClass("d-none");
+    }
```

Comment:
Wouldn't using jQuery's `show` / `hide` / `toggle` be simpler than manipulating classes?